src/components/PostDetails.tsx

```diff
@@ -1,107 +1,122 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment, getComments } from '../api/api';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+export interface Props {
+  post: Post;
+}
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [showForm, setShowForm] = useState(false);
 
-        <div className="block">
-          <Loader />
+  const title = `#${post.id}: ${post.title}`;
+  const showButton = !isLoading && !showForm && !isError;
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+    getComments(post)
+      .then(setComments)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  const handleChangeButton = () => {
+    setShowForm(true);
+  };
 
-          <p className="title is-4">Comments:</p>
+  const handleAddComment = (v: Comment) => {
+    setComments([...comments, v]);
+  };
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+  const handleDeleteComment = async (com: Comment) => {
+    setComments(prev => prev.filter(e => e.id !== com.id));
+    deleteComment(com);
```

Comment:
Consider awaiting the deleteComment function to ensure that the comment is successfully deleted before updating the state. This will help prevent inconsistencies if the deletion fails.